Doesn't build on arm64 linux

Linux debian-4gb-fsn1-1 6.1.0-11-arm64 #1 SMP Debian 6.1.38-4 (2023-08-08) aarch64 GNU/Linux

```
error[E0308]: mismatched types
   --> /home/mozgiii/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ls-qpack-0.1.2/src/decoder.rs:374:31
    |
374 |             this.header.buf = this.decoding_buffer.as_mut_ptr() as *mut i8;
    |             ---------------   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `*mut u8`, found `*mut i8`
    |             |
    |             expected due to the type of this binding
    |
    = note: expected raw pointer `*mut u8`
               found raw pointer `*mut i8`

error[E0308]: mismatched types
   --> /home/mozgiii/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ls-qpack-0.1.2/src/header.rs:199:18
    |
199 |             buf: header.buffer.as_mut_ptr() as *mut i8,
    |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `*mut u8`, found `*mut i8`
    |
    = note: expected raw pointer `*mut u8`
               found raw pointer `*mut i8`
```

I'm not sure why is this.

--- a/ls-qpack/src/decoder.rs
+++ b/ls-qpack/src/decoder.rs
@@ -275,6 +275,7 @@ impl Drop for InnerDecoder {
 mod callbacks {
     use crate::header::HeaderError;
     use crate::Header;
+    use std::ffi::c_char;
     use std::marker::PhantomPinned;
     use std::pin::Pin;
 
@@ -371,7 +372,7 @@ mod callbacks {
             this.decoding_buffer
                 .resize(space as usize, Default::default());
 
-            this.header.buf = this.decoding_buffer.as_mut_ptr() as *mut i8;
+            this.header.buf = this.decoding_buffer.as_mut_ptr() as *mut c_char;
             this.header.val_len = space;
         }
 

--- a/ls-qpack/src/header.rs
+++ b/ls-qpack/src/header.rs
@@ -2,6 +2,7 @@
 
 use std::convert::TryFrom;
 use std::fmt::Debug;
+use std::ffi::c_char;
 
 /// Error during Header construction.
 pub enum HeaderError {
@@ -196,7 +197,7 @@ impl<'a> LsxpackHeader<'a> {
         debug_assert!(header.value_len < Header::MAX_LEN_VALUE);
 
         let header_sys = ls_qpack_sys::lsxpack_header {
-            buf: header.buffer.as_mut_ptr() as *mut i8,
+            buf: header.buffer.as_mut_ptr() as *mut c_char,
             name_offset: header.name_offset as i32,
             name_len: header.name_len as u16,
             val_offset: header.value_offset as i32,

